mipengine/controller/node_registry.py

```diff
@@ -1,236 +1,32 @@
-
-        Parameters
-        ----------
-        node_id: the id of the node
-        data_model: the data model of the datasets
-        wanted_datasets: the datasets to look for
-
-        Returns
-        -------
-        some, all or none of the wanted_datasets that are located in the node
-        """
-
-        for node in self.nodes:
-            if node.id == node_id:
-                if data_model not in node.datasets_per_data_model.keys():
-                    raise ValueError(
-                        f"Data model '{data_model}' is not available in the node '{node_id}'."
-                    )
-                node_datasets = node.datasets_per_data_model[data_model]
-                break
-        else:
-            raise ValueError(
-                f"Node '{node_id}' could not be found in the node registry."
-            )
-
-        return list(set(node_datasets).intersection(wanted_datasets))
-
-    # returns a list of all the currently available data_models on the system
-    # without duplicates
-    def get_all_available_data_models(self) -> List[str]:
-        all_local_nodes = self.get_all_local_nodes()
-        tmp = [local_node.datasets_per_data_model for local_node in all_local_nodes]
-        all_existing_data_model = set().union(*tmp)
-        return list(all_existing_data_model)
-
-    # returns a dictionary with all the currently available data_models on the
-    # system as keys and lists of datasets as values. Without duplicates
-    def get_all_available_datasets_per_data_model(self) -> Dict[str, List[str]]:
-        all_local_nodes = self.get_all_local_nodes()
-        tmp = [node_info.datasets_per_data_model for node_info in all_local_nodes]
-        from collections import defaultdict
-        from itertools import chain
-        from operator import methodcaller
-
-        dd = defaultdict(list)
-        dict_items = map(methodcaller("items"), tmp)
-        for k, v in chain.from_iterable(dict_items):
-            dd[k].extend(v)
-
-        return dict(dd)
-
-    def data_model_exists(self, data_model: str):
-        for node_info in self.get_all_local_nodes():
-            if not node_info.datasets_per_data_model:
-                continue
-            if data_model in node_info.datasets_per_data_model.keys():
-                return True
-        return False
+        }
 
-    def dataset_exists(self, data_model: str, dataset: str):
-        for node_info in self.get_all_local_nodes():
-            if not node_info.datasets_per_data_model:
-                continue
-            if data_model not in node_info.datasets_per_data_model:
-                continue
-            if not _have_common_elements(
-                node_info.datasets_per_data_model[data_model], [dataset]
-            ):
-                continue
-            return True
-        return False
+    def get_node_info(self, node_id: str) -> NodeInfo:
+        return self.nodes[node_id]
 
 
 node_registry = NodeRegistry()
```

Comment:
What I mentioned in [mipengine/controller/data_model_registry.py](https://github.com/madgik/MIP-Engine/pull/200/files#diff-2a863839a93d4270ba5a37a6d8298524e331da01d8525a30b203ffaaf10d9de1) goes for the NodeRegistry class as well, it should be singleton and expose its inner attributes through a property